### PR TITLE
feat: CHARTS-7418 Add embedding examples for realm-web package

### PR DESCRIPTION
**Pull request by khanguslee**
With the release of `v2.3.0` we can now use the `realm-web` package to login using the `getRealmUserToken` method.

This PR adds both embedded charts and dashboard examples that use the `realm-web` package. Note that the old package `mongodb-stitch-browser-sdk` is now deprecated and `realm-web` is recommended now.

Other changes made in this PR include:
- Update any mentions of Realm to either Realm Web (SDK) or App Services (Back-end) found in code or documentation
- Update documentation with the new data sources page found on Atlas Charts
- Updated all `package.json` files with `v2.3.0` as the minimum version

### CODE REVIEW

**Comment by khanguslee** [comment]
It says 2 dead links found in https://github.com/mongodb-js/charts-embed-sdk/actions/runs/3141020990/jobs/5103015749 but these links will be created once this PR is merged.

File: examples/charts/authenticated-google/README.md [hunk 1]
```diff
@@ -39,30 +39,48 @@ This sample is pre-configured with a MongoDB-owned test Client ID. To use Google
...
+5. Make sure the "External users can view data in this data source" option is toggled on and "Allow authenticated data access" has been selected.
+
+### Enable chart for embedded access
+
+6. Go to the Dashboards tab on the side panel, and select the dashboard that contains the chart you wish to embed.
+
+7. Find the chart you want to embed, click the **...** menu and select **Embed chart**
 
-5. Go to the Dashboards tab on the left navigation column, and select the dashboard that contains the chart you wish to embed.
+8. Ensure the Unauthenticated tab is selected and turn on '**Enable Authenticated access**'
```
Comment (mongo-J) [inline examples/charts/authenticated-google/README.md:66]: Just to match the actual print
```suggestion
8. Ensure the **Authenticated** tab is selected and turn on '**Enable authenticated access**'
```

File: examples/charts/authenticated-realm-web/README.md [hunk 2]
```diff
@@ -10,16 +10,16 @@ MongoDB Charts allows you to create visualizations of your MongoDB data using a
 
 Charts can be embedded either using a simple IFRAME snippet, or by using the Charts Embedding SDK from your JavaScript code. When using the SDK, embedded charts can be either unauthenticated (meaning anyone who has the embed code can view the chart), or authenticated (whereby the user can only view the chart if they have an active authentication session linked to a Charts authentication provider).
```
Comment (mongo-J) [inline examples/charts/authenticated-realm-web/README.md:11]:
```suggestion
Charts can be embedded either using a simple IFRAME snippet, or by using the Charts Embedding SDK from your JavaScript code. When using the SDK, embedded charts can be either unauthenticated (meaning anyone who has access to the page where the chart is embedded can view the chart), or authenticated (whereby the user can only view the chart if they have an active authentication session linked to a Charts authentication provider).
```

File: examples/charts/authenticated-realm-web/README.md [hunk 3]
```diff
@@ -83,60 +83,75 @@ Choose or create a Realm Cloud app which will be used to authenticate users who
...
-     - This setting should be left off if you only require Realm to authenticate valid users, and you have no desire to filter data by their role in your Realm App. i.e, all valid users see the same chart, invalid users cannot see the chart.
-   - Enter the Realm Service name that retrieves your data. To find the name of this,
-     - Navigate to Clusters on the left hand navigation panel.
-     - Look for the Stitch Service Name associated with your Atlas cluster.
-       ![Screen Shot 2020-04-14 at 4 51 30 pm](https://user-images.githubusercontent.com/19422770/79194609-3478f280-7e70-11ea-9267-a1c7d35aad09.png)
+7. Find the chart you want to embed, click the **...** menu and select **Embed chart**
 
-Your provider config, (ignoring the completed optional settings) should look something like this.
+8. Ensure the Unauthenticated tab is selected and turn on '**Enable Authenticated access**'
```
Comment (mongo-J) [inline examples/charts/authenticated-realm-web/README.md:112]:
```suggestion
8. Ensure the Authenticated tab is selected and turn on '**Enable authenticated access**'
```

File: README.md [hunk 4]
```diff
@@ -34,7 +34,7 @@ yarn add @mongodb-js/charts-embed-dom
 
 - [Getting started with the MongoDB Charts SDK](https://github.com/mongodb-js/charts-embed-sdk/tree/master/examples/charts/unauthenticated)
 - [Getting started with custom JWT authentication](https://github.com/mongodb-js/charts-embed-sdk/tree/master/examples/charts/authenticated-custom-jwt)
-- [Getting started with Realm authentication](https://github.com/mongodb-js/charts-embed-sdk/tree/master/examples/charts/authenticated-realm)
+- [Getting started with Realm Web authentication](https://github.com/mongodb-js/charts-embed-sdk/tree/master/examples/charts/authenticated-realm-web)
```
Comment (mongo-J) [inline README.md:37]: I wonder to what extent we would like to replace realm web with `App Services`, I know the npm package name is called `realm-web`, should we keep it just at npm level and refer everything else as App services?

**Review by mongo-J — COMMENTED** [review]
Thanks for updating the examples and updating the instructions! Just a few nits on namings

File: examples/charts/authenticated-google/README.md (hunk 1 above)
Comment (kristinamongo) [inline examples/charts/authenticated-google/README.md:66]:
```suggestion
8. Ensure the Authenticated tab is selected and turn on '**Enable Authenticated access**'
```

File: examples/charts/authenticated-realm-deprecated/README.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+5. Make sure the "External users can view data in this data source" option is toggled on and "Allow authenticated data access" has been selected.
+
+### Enable chart for embedded access
+
+6. Go to the Dashboards tab on the left navigation column, and select the dashboard that contains the chart you wish to embed.
+
+7. Find the chart you want to embed, click the **...** menu and select **Embed chart**
+
+8. Ensure the Unauthenticated tab is selected and turn on '**Enable Authenticated access**'
```
Comment (kristinamongo) [inline examples/charts/authenticated-realm-deprecated/README.md:114]:
```suggestion
8. Ensure the Authenticated tab is selected and turn on '**Enable Authenticated access**'
```

File: examples/charts/authenticated-realm-web/README.md [hunk 6]
```diff
@@ -10,16 +10,16 @@ MongoDB Charts allows you to create visualizations of your MongoDB data using a
```
Comment (kristinamongo) [inline examples/charts/authenticated-realm-web/README.md:10]: I think the link on line 7 be changed to `...realm-web` too

File: README.md (hunk 4 above)
Comment (kristinamongo) [reply README.md:37]: Tricky question 😬 Would it be ok if we have it with both names here as this `readme` is the gateway to all examples and is the thing which everyone sees when they come to the repo. Everywhere else, we can refer to it as `App Services`. So `Getting started with App Services (Realm Web) authentication`. What do you think?

Comment (mongo-J) [reply README.md:37]: > How about `"Getting started with App Services authentication via Realm Web SDK"`?

This sounds like a good idea 👍

File: examples/charts/authenticated-realm-web/README.md (hunk 6 above)
Comment (khanguslee) [reply examples/charts/authenticated-realm-web/README.md:10]: good catch! You're a life saver 😄

**Review by kristinamongo — APPROVED** [review]
Looks great! Thanks for fixing all docs ❤️

**Pull request merged**